pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -149,8 +151,8 @@ func (m *ManagerImpl) genericDeviceUpdateCallback(resourceName string, devices [
 			m.unhealthyDevices[resourceName].Insert(dev.ID)
 		}
 	}
-	m.mutex.Unlock()
 	m.writeCheckpoint()
```

Comment:
Can you explain why writeCheckpoint() needs to be protected with mutex lock? I think currently, we take the mutex lock when we generate checkpoint data inside writeCheckpoint(), and then at the checkpointManager level, it has its own lock to update checkpoint file.

Reply:
Ah, I see the mutex locking in createCheckpoint() within the checkpointManager itself, since it appeared that writeCheckpoint() didn't actually lock the checkpoint creation. Thanks for pointing this out.

Initially I was thinking that callback appears to be the initial point where the checkpoint is create and written and that thought while we are trying to list the checkpoints in line 845, it could be the case that the checkpoint would be in the process of being created.